giskard/llm/evaluators/string_matcher.py

```diff
@@ -0,0 +1,103 @@
+import re
+import string
+import logging
+from typing import Tuple, Dict, List
+from dataclasses import dataclass
+
+from .base import BaseEvaluator, EvaluationResult
+from ...datasets.base import Dataset
+from ...models.base.model import BaseModel
+from ..errors import LLMGenerationError
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+@dataclass(frozen=True)
+class StringMatchingMethod:
+    substrings: Tuple[str]
```

Comment:
I think that now that you’ve added all different matching methods we can rename this to something like just `strings`/`target_strings`/`expected_matches` or similar, as they are not necessarily substrings (e.g. for exact match).